### ISSUE TITLE
gemma4: the cacheless and cache-bearing text forwards disagree, and the gap grows with attended context

### ISSUE THREAD

**Issue by Pushkinist** [Pushkinist/rMLX#495]
## Summary

Gemma4's two text forwards disagree, and the gap grows with the attended
context. `Gemma4Text::forward_seq_logits_all` (one pass over the whole window,
no cache) and `Gemma4Text::forward_seq_with_cache` (prefill + single-token
steps) are supposed to compute the same thing for the same token ids. They do,
exactly, for very short windows. Past that they diverge, monotonically, and by
a margin far outside floating-point noise.

Qwen3 does not do this, which is what makes it a Gemma4 finding rather than a
property of comparing a batched forward with a sequential one.

## Measured

`rmlx eval ppl` against itself: `--kv-quant none` (a bf16 cache — no
quantization anywhere) versus no KV flag (the cacheless scorer). Same corpus
prefix, same token ids, same scored positions, `--stride == --ctx-window`,
`--max-tokens 512`, wikitext-2 raw, temp 0, deterministic. The only difference
is which forward produced the logits.

`mean_nll`, cached minus cacheless:

| `--ctx-window` | windows | scored | gemma-4-12B-it-mxfp8 | Ternary-Bonsai-8B (Qwen3) |
|---:|---:|---:|---:|---:|
| 8 | 64 | 448 | **0.000000** | **0.000000** |
| 16 | 32 | 480 | +0.011885 | −0.516963 |
| 32 | 16 | 496 | −0.048008 | +0.003221 |
| 64 | 8 | 504 | −0.105303 | +0.001431 |
| 128 | 4 | 508 | −0.090060 | +0.000869 |
| 256 | 2 | 510 | −0.114481 | +0.000991 |
| 512 | 1 | 511 | **−0.122849** | +0.000440 |

At `--ctx-window 2` and `4` the two are bit-identical on both architectures
(`11.8077965733` twice, `13.0235197855` twice on 12B), which is the control:
the harness, the token ids, the scored set and the NLL accumulation are shared,
so a difference cannot come from them.

At `--ctx-window 512` on a model whose `sliding_window` is 1024, the
sliding-attention band covers the whole window — so this is **not** the
rotating ring against the prefill SWA mask. The residual is already −0.105 at
`--ctx-window 64`.

`gemma-4-26b-a4b-it-mxfp8` shows the same shape (`mean_nll` 10.9276 cacheless
against 10.8593 cached at `--ctx-window 1024`).

## Why it matters

* Every `ppl_*` row in `runs.db` for a Gemma4 model was produced by the
  cacheless forward. If the cached one is the correct one — and it is the one a
  served request runs — those numbers are not the checkpoint's perplexity.
* A Gemma4 quality comparison that mixes the two is meaningless, which is why
  `docs/CLI.md` now says the two scorers are not interchangeable on Gemma4 and
  the cached scorer records under its own metric.

## Where to look

`crates/rmlx-models/src/gemma4/model.rs:207` (`forward_seq_logits_all`) passes
`cache = None`, `attn_is_rotating = false` and `offset = 0` to every layer;
`crates/rmlx-models/src/gemma4/layers/mod.rs:369` is the no-cache branch that
then sizes its mask from the model-wide offset. The cached path takes the
`Some(cache)` branch above it, with `producer_effective_offset` and the
rotating short-circuit. Those two branches are the candidate.

The independent variable to bisect on is the attended context, not the window
count: the residual at one window (`--ctx-window 512`, 1 window) is the largest
in the table.

## Not in scope of the PR that found it

This was surfaced by #494, which put the two forwards side by side for the
first time by giving `rmlx eval ppl` a cache-bearing mode. Nothing in that PR
changes either forward, and the cross-mode control there is what produced the
table above. `score_window_through_cache` is excluded by the `--ctx-window 8`
row and by the Qwen3 column.

## Reproduce

```
scripts=.rmlx/analysis/443/run2
bash $scripts/crossmode_sweep.sh      # the table above
bash $scripts/crossmode_tiny.sh       # the bit-identical control at ctx 2/4
```
